### PR TITLE
Add samtools-depth

### PR DESCRIPTION
**Pull request by daynefiler**
Adds samtools depth wrapper. Test added to test.py.

### CODE REVIEW

File: bio/samtools/depth/test/Snakefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+rule samtools_depth:
+    input:
+        bams = ["mapped/A.bam", "mapped/B.bam"],
+    output:
+        "depth.txt"
+    params:
+        # optional bed file passed to -b
+        bed = "regionToCalcDepth.bed",
+        extra = "" # optional additional parameters as string
```
Comment (johanneskoester) [inline bio/samtools/depth/test/Snakefile]: Same here, please add as optional input files.

Comment (daynefiler) [reply bio/samtools/depth/test/Snakefile]: When I had them as inputs and did not provide a file, e.g. bed = "" I was getting input rule exception errors. How do I designate optional inputs?

**Comment by daynefiler** [comment]
I updated as requested, using the wrapper from one of the gatk modules as a guide.

File: bio/samtools/depth/test/Snakefile (hunk 1 above)
Comment (johanneskoester) [reply bio/samtools/depth/test/Snakefile]: Just allow to comment them out.

File: bio/samtools/depth/test/Snakefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+rule samtools_depth:
+    input:
+        bams = ["mapped/A.bam", "mapped/B.bam"],
+        bed = "regionToCalcDepth.bed", # optional
```
Comment (johanneskoester) [inline bio/samtools/depth/test/Snakefile]: Please follow Python style, no whitespace around keyword args.

**Comment by daynefiler** [comment]
Sorry for the delayed response -- updated w/o spaces.

**Pull request merged**